sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/AesKw.cs

```diff
@@ -0,0 +1,422 @@
+
+                if (inputOffset < 0)
+                    throw new ArgumentOutOfRangeException(nameof(inputOffset), "The inputOffset parameter must not be negative");
+
+                if (inputOffset + inputCount > inputBuffer.Length)
+                    throw new ArgumentOutOfRangeException(nameof(inputCount), "The sum of inputCount and inputOffset parameters must not be larger than the length of inputBuffer");
+
+                /*
+                   1) Initialize variables.
+
+                       Set A = IV, an initial value (see 2.2.3)
+                       For i = 1 to n
+                           R[i] = P[i]
+
+                   2) Calculate intermediate values.
+
+                       For j = 0 to 5
+                           For i=1 to n
+                               B = AES(K, A | R[i])
+                               A = MSB(64, B) ^ t where t = (n*j)+i
+                               R[i] = LSB(64, B)
+
+                   3) Output the results.
+
+                       Set C[0] = A
+                       For i = 1 to n
+                           C[i] = R[i]
+                */
+
+                // The default initialization vector from RFC3394
+                byte[] a = _iv;
+
+                // The number of input blocks
+                var n = inputCount >> 3;
+
+                // The set of input blocks
+                byte[] r = new byte[n << 3];
+
+                Array.Copy(inputBuffer, inputOffset, r, 0, inputCount);
+
+                var encryptor = _aes.CreateEncryptor();
+                byte[] block = new byte[16];
+
+                // Calculate intermediate values
+                for (var j = 0; j < 6; j++)
+                {
+                    for (var i = 0; i < n; i++)
+                    {
+                        // T = ( n * j ) + i
+                        var t = (ulong)((n * j) + i + 1);
+
+                        // B = AES( K, A | R[i] )
+
+                        // First, block = A | R[i]
+                        Array.Copy(a, block, a.Length);
+                        Array.Copy(r, i << 3, block, 64 >> 3, 64 >> 3);
+
+                        // Second, AES( K, block )
+                        var b = encryptor.TransformFinalBlock(block, 0, 16);
+
+                        // A = MSB( 64, B )
+                        Array.Copy(b, a, 64 >> 3);
+
+                        // A = A ^ t
+                        a.Xor(GetBytes(t), true);
+
+                        // R[i] = LSB( 64, B )
+                        Array.Copy(b, 64 >> 3, r, i << 3, 64 >> 3);
+                    }
+                }
+
+                var c = new byte[(n + 1) << 3];
+
+                Array.Copy(a, c, a.Length);
+
+                for (var i = 0; i < n; i++)
+                {
+                    Array.Copy(r, i << 3, c, (i + 1) << 3, 8);
+                }
+
+                return c;
+            }
+
+            public void Dispose()
```

Comment:
Shouldn't you have a finalizer here?

Reply:
A finalizer shouldn't be necessary here since it doesn't directly hold and unmanaged resources.  Aes already has a finalizer.